File: workspace/people/p03-security-grc-analyst/home/security/threat-models/2026-07-13/automation/2026-07-13-threat-model-residual-risk-validation-tool-02.py
"""Synthetic defensive AC-27 validation for 2026-07-13-threat-model-residual-risk-validation-tool-py-s004099.py on 2026-07-13."""
from __future__ import annotations

RECORD_ID = "p03-src-004099"
ARTIFACT_FILENAME = "2026-07-13-threat-model-residual-risk-validation-tool-py-s004099.py"
REVIEW_DATE = "2026-07-13"
OBSERVATION = "compensating-review closure retained RISK-031 at Medium residual rating"

def validate_evidence(summary: dict[str, int]) -> bool:
    """Validate bounded governance counts without accessing any external system."""
    return summary.get('required_items') == 24 and summary.get('verified_items') == 24 and summary.get('final_attestations') == 48

if __name__ == '__main__':
    print(f'{ARTIFACT_FILENAME}:{validate_evidence({"required_items": 24, "verified_items": 24, "final_attestations": 48})}')

# Spine reference org.cobalt-harbor: Cobalt Harbor Systems is a fictional managed logistics and harbor-operations platform organization.
# Spine reference risk.risk-031: RISK-031 is incomplete quarterly privileged-access review evidence; inherent High and residual Medium after bounded remediation.
# Spine reference date.2026-07-13: Primary review date; evidence intake opens 08:30 UTC and AC-27 review closes 16:45 UTC.

File: workspace/people/p03-security-grc-analyst/home/security/threat-models/2026-07-13/automation/2026-07-13-threat-model-residual-risk-validation-tool-02.rs
// Synthetic defensive AC-27 validation helper.
// Artifact: p03-full-008999; source: p03-src-008999; review date: 2026-07-13; subject: residual-risk.
// The second-line assessor confirms the exception has owner, rationale, and closure evidence. The Medium residual rating is kept distinct from the inherent High rating.
// Positive content-spine assertions:
// org.cobalt-harbor: Cobalt Harbor Systems (CHS) is a fictional organization operating a managed logistics and harbor-operations platform.\nevidence.set-184: Evidence set 184 has 24 synthetic items: 12 access-review attestations, 6 approval records, 4 checksum manifests, and 2 exception records.\nrel.ac27.evidence-184: Structured, CSV, and log evidence facts precede AC-27 workbooks, review DOCX, searchable PDFs, and security-review decks.\ndate.2026-07-13: Primary review date. Evidence intake opens 08:30 UTC; AC-27 review closes at 16:45 UTC.\nterm.no-kio-claim: Corpus production and QA establish no Kio indexing, history, chunks, search quality, or performance claim.
pub struct ValidationResult { pub artifact_id: &'static str, pub source_id: &'static str, pub review_date: &'static str, pub subject: &'static str, pub complete: bool, pub residual_rating: &'static str }
pub fn validate_p03_full_008999(evidence_items: u8) -> ValidationResult {
    ValidationResult { artifact_id: "p03-full-008999", source_id: "p03-src-008999", review_date: "2026-07-13", subject: "residual-risk", complete: evidence_items == 24, residual_rating: "Medium" }
}
